zilliqa/src/consensus.rs

```diff
@@ -279,28 +235,45 @@ impl Consensus {
             }
         };
 
-        let (start_view, high_qc) = {
+        let (start_view, finalized_view, high_qc) = {
             match db.get_high_qc()? {
                 Some(qc) => {
                     let high_block = block_store
                         .get_block(qc.block_hash)?
                         .ok_or_else(|| anyhow!("missing block that high QC points to!"))?;
 
                     let finalized_number = db
-                        .get_latest_finalized_view()?
+                        .get_finalized_view()?
                         .ok_or_else(|| anyhow!("missing latest finalized view!"))?;
                     let finalized_block = db
                         .get_block_by_view(finalized_number)?
                         .ok_or_else(|| anyhow!("missing finalized block!"))?;
 
-                    let start_view = std::cmp::max(high_block.view(), finalized_block.view()) + 1;
+                    // If latest view was written to disk then always start from there. Otherwise start from (highest out of high block and finalised block) + 1
+                    let mut start_view = db
+                        .get_view()?
+                        .or_else(|| {
+                            Some(std::cmp::max(high_block.view(), finalized_block.view()) + 1)
+                        })
+                        .unwrap();
+
                     trace!(
                         "recovery: high_block view {0}, finalized_number {1} , start_view {2}",
                         high_block.view(),
                         finalized_number,
                         start_view
                     );
 
+                    // If timestamp of when current view was written exists then use it to estimate the minimum number of blocks the network has moved on since shut down
```

Comment:
I think when doing this catch up we should also send the `NewView`s for the missed views that we didn't send. Otherwise we still need to wait for the next view to progress before the nodes will re-sync.

We could discuss and make that change in a separate PR if you prefer.

Reply:
Will create a separate issue.